fire/SFMainMod.F90

```diff
@@ -679,6 +684,14 @@ subroutine  area_burnt_intensity ( currentSite )
     real(r8) AB               !daily area burnt in m2 per km2
     
     real(r8) size_of_fire !in m2
+    real(r8) cloud_to_ground_strikes  ! [fraction] depends on hlm_spitfire_mode
+    integer :: iofp  ! index of oldest fates patch
+    ! The following three parameters represent values of hlm_spitfire_mode
+    ! NB. The same parameters are set in /src/biogeochem/CNFireFactoryMod
+    integer, parameter :: scalar_lightning = 1  ! value of scalar_lightning mode
```

Comment:
@slevisconsulting I am not sure what to call this given the way this logic is being used. It is acting as a dual switch for either the default constant ignitions and the lightning ignitions, so it is harder to name...

Reply:
The way I see it @jkshuman 
`hlm_spitfire_mode = 1`  corresponds to the global constant lightning case (no dataset). This is why I named it `scalar_lightning`. It may help to add one more parameter:
`integer, parameter :: lightning_from_dataset = 2  ! value of lightning_from_dataset mode`
My if-statements didn't need that one, but I could add it for clarity if you recommend it.

Reply:
If so, I'm adding a reminder here to rename the corresponding parameter in https://github.com/ESCOMP/CTSM/pull/991/files#diff-66b3ac7e716691e7e348de97bd5cba79
(line 150) in a consistent manner.

Reply:
@slevisconsulting including the  would make things quite clear. I wish there was a better way to sync the connection between the spitfire_modes and this section rather than hard-coding these numbers. It just seems like it opens a door to a missed connection if someone changes things in one section and not in the HLM portion. (I can't think of any suggestions though.)

Reply:
I have now made these parameters public to avoid declaring them twice.

Reply:
By the way, thank you for all your comments in these two PRs @jkshuman .